distribution/sdk/sample-mpegts-streamgenerator/src/main/java/org/codice/alliance/distribution/sdk/video/stream/mpegts/MpegTsUdpClient.java

```diff
@@ -182,11 +192,37 @@ public static void main(String[] args) {
                 tsDurationMillis,
                 minDatagramSize,
                 maxDatagramSize,
-                fractionalTs);
+                fractionalTs,
+                networkInterface);
+    }
+
+    private static Optional<InetAddress> findLocalAddress(String interfaceName) {
+
+        try {
+
+            NetworkInterface networkInterface = NetworkInterface.getByName(interfaceName);
+
+            if (networkInterface != null) {
+                return Collections.list(networkInterface.getInetAddresses())
+                        .stream()
+                        .filter(inetAddress -> inetAddress instanceof Inet4Address)
+                        .findFirst();
+            }
+
+        } catch (SocketException e) {
+            LOGGER.info("unable to find the network interface {}", interfaceName, e);
+        }
+
+        return Optional.empty();
     }
 
     public static void broadcastVideo(String videoFilePath, String ip, int port,
-            long tsDurationMillis, int minDatagramSize, int maxDatagramSize, boolean fractionalTs) {
+            long tsDurationMillis, int minDatagramSize, int maxDatagramSize, boolean fractionalTs,
+            String networkInterfaceName) {
+
+        Optional<InetAddress> inetAddressOptional =
+                networkInterfaceName == null ? Optional.empty() : findLocalAddress(
```

Comment:
Would it be clearer to pass `networkInterfaceName` to `findLocalAddress` no matter what and have `findLocalAddress` return `Optional.empty()` if it's `null`?